scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,279 @@
+const path = require('path');
+const { Parser } = require('@asyncapi/parser');
+const parser = new Parser();
+
+// Read the markdown file
+const markdownContent = fs.readFileSync('../../spec/asyncapi.md', 'utf8');
+
+// Function to extract comments with example metadata
+function extractComments(content) {
+  const commentRegex = /<!-- asyncapi-example-tester:(\{.*?\}) -->/g;
+  let match;
+  const comments = [];
+
+  while ((match = commentRegex.exec(content)) !== null) {
+    try {
+      comments.push({
+        json: JSON.parse(match[1]),
+        index: match.index
+      });
+    } catch (e) {
+      console.error("Failed to parse comment JSON:", match[1], e);
+    }
+  }
+
+  return comments;
+}
+
+// Function to extract JSON examples from markdown content
+function extractExamples(content) {
+  const exampleRegex = /```json\s+([\s\S]*?)\s+```/g;
+  let match;
+  const examples = [];
+
+  while ((match = exampleRegex.exec(content)) !== null) {
+    examples.push({
+      json: match[1],
+      index: match.index
+    });
+  }
+
+  return examples;
+}
+
+// Extract comments from the markdown file
+const comments = extractComments(markdownContent);
+// Extract examples from the markdown file
+const examples = extractExamples(markdownContent);
+
+// Create array of objects with properties: 'name', 'json_path', 'example'
+const combinedData = comments.map((comment) => {
+  const matchingExample = examples.find(example => example.index > comment.index);
+  if (matchingExample) {
+    try {
+      return {
+        name: comment.json.name,
+        json_path: comment.json.json_path,
+        example: JSON.parse(matchingExample.json)
+      };
+    } catch (e) {
+      console.error("Failed to parse example JSON:", matchingExample.json, e);
+      return null;
+    }
+  } else {
+    console.error(`No matching example found for comment: '${comment.json.name}'`);
+    return null;
+  }
+}).filter(item => item !== null);
+
+// Function to deeply merge two objects without overwriting existing nested structures
+function deepMerge(target, source) {
+  for (const key of Object.keys(source)) {
+    if (source[key] instanceof Object && key in target && target[key] instanceof Object) {
+      target[key] = deepMerge(target[key], source[key]);
+    } else {
+      target[key] = source[key];
+    }
+  }
+  return target;
+}
+
+// Function to set a value in a JSON object using JSONPath, creating missing fields if necessary
+function setValueByPath(obj, path, value) {
+  const pathParts = path.replace(/\$/g, '').split('.').slice(1); // Remove the root "$" and split path
+  let current = obj;
+
+  pathParts.forEach((part, index) => {
+    if (index === pathParts.length - 1) {
+      current[part] = value; // Set the new value directly
+    } else {
+      if (!current[part]) {
+        current[part] = {}; // Create object if it doesn't exist
+      }
+      current = current[part];
+    }
+  });
+}
+
+// Function to apply updates and save the document
```

Comment:
Why should you save to disk?

Reply:
> Why should you save to disk?

@smoya 
So I thought of this approach:
Saving temporarily --> validation of all examples --> print result --> delete the temp files

Reply:
But what's the value of writing to disk, if there is one?

Reply:
> But what's the value of writing to disk, if there is one?

@smoya 
Let me explain a bit more.
- The parser validation will be [async in nature](https://github.com/asyncapi/spec/blob/ed557896c4a7d9dae334f4515efc3cbf6118c784/scripts/validation/embedded-examples-validation.js#L190C1-L210C1), so I thought it would be better to collect every example and write that down in a separate folder. 
- Then run the validation for that whole folder to get every example checked.

Do you have perhaps any other suggestions for this?

Reply:
Why not having everything in memory instead of writing to disk?

Reply:
> Why not having everything in memory instead of writing to disk?

Let me try that...

Reply:
resolved via e2cc95a